### PR TITLE
Add autoregistration module

### PR DESCRIPTION
**Pull request by masa-orca**
##### SUMMARY

- Add zabbix_autoregistration module (Update only).
- Add tests about zabbix_autoregistration module.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

zabbix_autoregistration

##### ADDITIONAL INFORMATION

I create PR for  a module which mentioned at a [issue](https://github.com/ansible-collections/community.zabbix/issues/627).
I am not good at English. Please commnet if I need to change words and grammars!

### CODE REVIEW

**Comment by rockaut** [comment]
Please see my last comment on the issue. Sorry for late response.

This function is a part of zabbix action already.

Please have a look at this and feel free to ask us if you need help.

**Comment by rockaut** [comment]
Sorry,hit the wrong button

File: plugins/modules/zabbix_autoregistration.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+            - Type of allowed incoming connections for autoregistration.
+            - Choose from C(unsecure), C(tls_with_psk) or both.
+        type: list
+        elements: str
+        required: true
+    tls_psk_identity:
+        description:
+            - TLS connection use this PSK identity string.
+            - The PSK identity string will be transmitted unencrypted over the network. Therefore, you should not put sensitive information.
+            - It is required if you set C(tls_with_psk) to I(tls_accept) and current tls_accept is C(unsecure).
```
Comment (D3DeFi) [inline plugins/modules/zabbix_autoregistration.py:39]:
```suggestion
            - TLS connection uses this PSK identity string.
            - The PSK identity string will be transmitted unencrypted over the network. Therefore, you should not put any  sensitive information here.
            - This setting requires I(tls_accept=tls_with_psk) if current value of I(tls_accept) is C(unsecure).
```

File: plugins/modules/zabbix_autoregistration.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        type: str
+    tls_psk:
+        description:
+            - TLS connection use this PSK value.
+            - It is required if you set C(tls_with_psk) to I(tls_accept) and current tls_accept is C(unsecure).
```
Comment (D3DeFi) [inline plugins/modules/zabbix_autoregistration.py:44]:
```suggestion
            - TLS connection uses this PSK value.
            - This setting requires I(tls_accept=tls_with_psk) if current value of I(tls_accept) is C(unsecure)
```

Does the 2nd line mean that `tls_accept=tls_with_psk` is only required when you updating from `tls_accept=unsecure`, but can be omitted when re-running multiple times without module doing any changes itself?

File: plugins/modules/zabbix_autoregistration.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    login_password: secret
+    tls_accept: unsecure
+'''
+
+RETURN = '''
+msg:
+    description: The result of the operation
+    returned: success
+    type: str
+    sample: 'Successfully update autoregistration'
```
Comment (D3DeFi) [inline plugins/modules/zabbix_autoregistration.py:81]:
```suggestion
    sample: 'Successfully updated global autoregistration setting'
```

File: plugins/modules/zabbix_autoregistration.py [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+import ansible_collections.community.zabbix.plugins.module_utils.helpers as zabbix_utils
+
+
+class Autoregistration(ZabbixBase):
+    def __init__(self, module, zbx=None, zapi_wrapper=None):
+        super(Autoregistration, self).__init__(module, zbx, zapi_wrapper)
+        self.existing_data = None
+        if LooseVersion(self._zbx_api_version) < LooseVersion('4.4.0'):
+            module.fail_json(msg="This module unsuport Zabbix %s" % self._zbx_api_version)
```
Comment (D3DeFi) [inline plugins/modules/zabbix_autoregistration.py:98]:
```suggestion
            module.fail_json(msg="This module doesn't support Zabbix versions lower than %s" % self._zbx_api_version)
```

File: plugins/modules/zabbix_autoregistration.py [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+        type: str
+    tls_psk:
+        description:
+            - TLS connection use this PSK value.
+            - It is required if you set C(tls_with_psk) to I(tls_accept) and current tls_accept is C(unsecure).
+        type: str
+
+notes:
+    - Only Zabbix >= 4.4 is supported.
+    - This module returns state of changed is true when you set values to I(tls_psk_identity) and I(tls_psk).
```
Comment (D3DeFi) [inline plugins/modules/zabbix_autoregistration.py:49]:
```suggestion
    - This module returns changed=true when any value is set in I(tls_psk_identity) or I(tls_psk) as Zabbix API will not return any sensitive information back for module to compare.
    - Please note that this module configures B(global Zabbix Server settings). If you want to create autoregistration action so your hosts can automatically add themselves to the monitoring have a look at M(community.zabbix.zabbix_action).
```

**Review by D3DeFi — CHANGES_REQUESTED** [review]
Thank you for this contribution! :+1: :rocket: 

I took a liberty and corrected some of the grammar. Please see my suggestions. 

Also please add YAML drop-in file to changelog/fragments/630-zabbix-autoreg.yml with content like:
```
minor_changes:
- new module zabbix_autoregistration for configuring global autoregistration settings in Zabbix Server's Settings section of GUI
```
Otherwise very nice contribution

File: plugins/modules/zabbix_autoregistration.py (hunk 2 above)
Comment (masa-orca) [reply plugins/modules/zabbix_autoregistration.py]: I tried it. However, I could not come up with best way to get last task's arguments.
In addition, I think these values must be not logged.

**Comment by masa-orca** [comment]
@D3DeFi Thank you for your commens! and I have reflected your comments. Would you check my new commit?

**Review by D3DeFi — APPROVED** [review]
Very nice contribution :) +1 for merge.

@rockaut please decide which name we wsnt for the module and if this is fine with you, feel free to merge

**Comment by rockaut** [comment]
I haven't tested it personally - but it looks great for me. Thanks you @masa-orca and sorry for me screwing it up with the typo. +1 from me too.

**Comment by dj-wasabi** [comment]
Will merge it, as it seems that both @D3DeFi and @rockaut  agree. 😄

**Comment by D3DeFi** [comment]
Hmm, @masa-orca care to open follow-up PR for renaming? :D

**Comment by dj-wasabi** [comment]
Ok, my bad. I will try to do that somewhere in the next few days. No need for @masa-orca to do that.

**Pull request merged**